refactor: replace all calls to os.Exit() / CheckErr().

jfbai opened the pull request:
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Fix https://github.com/kubernetes/kubeadm/issues/1742
**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubernetes/kubeadm/issues/1742

**Special notes for your reviewer**:

None

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
None
```

/assign @rosti 
/assign @neolit123

jfbai committed:
refactor: replace all calls to os.Exit() / CheckErr().

jfbai commented:
/sig cluster-lifecycle
/area kubeadm

neolit123 commented on cmd/kubeadm/app/cmd/alpha/certs.go:
```diff
@@ -188,20 +192,24 @@ func addRenewFlags(cmd *cobra.Command, flags *renewFlags) {
 	internalcfg, err := configutil.LoadOrDefaultInitConfiguration(flags.cfgPath, &kubeadmapiv1beta2.InitConfiguration{}, &flags.cfg)
-	kubeadmutil.CheckErr(err)
+	if err != nil {
+		return
```
we might have to return the actual error from this function here.

jfbai replied:
Fixed.

neolit123 commented:
thanks @jfbai 
this is looking good thus far, i've added one comment.

btw, the PR needs to wait until the ~10th of Sept. because we are in code freeze for 1.16 ATM.

neolit123 commented:
/ok-to-test

jfbai commented:
> thanks @jfbai
> this is looking good thus far, i've added one comment.
> 
> btw, the PR needs to wait until the ~10th of Sept. because we are in code freeze for 1.16 ATM.

@neolit123 Thanks a lot, I have fixed your comment.

jfbai commented:
/test pull-kubernetes-e2e-gce-100-performance
/test pull-kubernetes-kubemark-e2e-gce-big

rosti reviewed (commented):
Thanks @jfbai !

I have just some style nits.
One additional thing, that we need to do is to add the following couple of lines, just before [this line](https://github.com/kubernetes/kubernetes/blob/c7c89f8c6140e0639dc8ba2d8dcdd6fdbd5409ae/cmd/kubeadm/app/cmd/cmd.go#L71):

```golang
SilenceErrors: true,
SilenceUsage: true,
```

This is because, the Cobra behavior changes slightly when `RunE` is used. If an error is returned it's dumped by Cobra and the command usage is displayed. Hence, to preserve the current behavior, we need to disable that.

jfbai commented:
> Thanks @jfbai !
> 
> I have just some style nits.
> One additional thing, that we need to do is to add the following couple of lines, just before [this line](https://github.com/kubernetes/kubernetes/blob/c7c89f8c6140e0639dc8ba2d8dcdd6fdbd5409ae/cmd/kubeadm/app/cmd/cmd.go#L71):
> 
> ```go
> SilenceErrors: true,
> SilenceUsage: true,
> ```
> 
> This is because, the Cobra behavior changes slightly when `RunE` is used. If an error is returned it's dumped by Cobra and the command usage is displayed. Hence, to preserve the current behavior, we need to disable that.

@rosti Thanks for comments, I will fix them. :).

PS: previously, I returned the error in one line as you commented. Then, I thought that check `err` may help understanding and stay the `if` blocks.

jfbai commented:
/test pull-kubernetes-bazel-build

jfbai commented:
Hi @rosti , I fixed your comments, could you please take another look?

rosti reviewed (commented):
Thanks @jfbai !
We just need to fix a few nits in the tests and this is ready to go (as soon as the master is unblocked again).

jfbai commented:
/test pull-kubernetes-node-e2e

jfbai commented:
/test pull-kubernetes-node-e2e

jfbai commented:
/test pull-kubernetes-node-e2e

jfbai commented:
@rosti PTAL, thanks a lot.

jfbai commented:
/test pull-kubernetes-node-e2e

neolit123 commented:
/priority backlog

jfbai commented:
@neolit123 @rosti Could this PR be merged now?

rosti reviewed (approved):
Looks great @jfbai !
Sorry for the delay, I was on a PTO.
/lgtm
/hold cancel
